### PR TITLE
Anko - Update Group data source

### PR DESCRIPTION
**Pull request by at-cuongcao**
#### Related issues (Required)
- Task #159 

#### Description (Required)
- [x] Update Group data source

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 1]
```diff
@@ -122,7 +131,15 @@ class GroupRemoteDataSource : GroupDataSource {
...
     override fun removeUserFromGroup(userId: String): Single<Boolean> {
-        TODO("not implemented")
+        val result = SingleSubject.create<Boolean>()
+        HypertrackApi.instance.removeUserFromGroup(userId, BodyAddUserToGroup(null))
+                .doOnSuccess { result.onSuccess(true) }
+                .doOnError { result.onError(it) }
+        return result
+    }
+
+    override fun searchGroup(groupName: String): Observable<SearchGroupResult> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/group/GroupActivityViewModel.kt [hunk 2]
```diff
@@ -12,20 +14,26 @@ import vn.asiantech.way.extension.observeOnUiThread
  */
 class GroupActivityViewModel {
 
-    private val groupDataSource = GroupRepository()
+    private val groupRepository = GroupRepository()
+    private val wayRepository = WayRepository()
+
+    internal fun getUser(): Observable<User> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupActivityViewModel.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 3]
```diff
@@ -28,26 +29,30 @@ class GroupRepository : GroupDataSource {
     }
 
     override fun getRequest(groupId: String): Observable<Invite> {
-        TODO("not implemented")
+        return remoteDataSource.getRequest(groupId)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 4]
```diff
@@ -28,26 +29,30 @@ class GroupRepository : GroupDataSource {
...
     }
 
     override fun upGroupInfo(group: Group): Observable<Boolean> {
-        TODO("not implemented")
+        return remoteDataSource.upGroupInfo(group)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 5]
```diff
@@ -28,26 +29,30 @@ class GroupRepository : GroupDataSource {
...
     }
 
-    override fun changeOwner(groupId: String, newOwner: String): Observable<Boolean> {
-        TODO("not implemented")
+    override fun changeOwner(groupId: String, newOwner: String): Single<Boolean> {
+        return remoteDataSource.changeOwner(groupId, newOwner)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/model/SearchGroupResult.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package vn.asiantech.way.data.model
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by cuongcaov on 05/12/2017
+ */
+data class SearchGroupResult(val results: MutableList<Group>)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/model/SearchGroupResult.kt]: `results` change to `groups`

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt (hunk 3 above)
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: `getRequest ` to `getGroupRequest`

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt (hunk 4 above)
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: `upGroupInfo` to `postGroupInfo`

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt (hunk 5 above)
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: `changeOwner` change to `changeGroupOwner`

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 7]
```diff
@@ -28,26 +29,30 @@ class GroupRepository : GroupDataSource {
...
     }
 
     override fun removeGroup(groupId: String): Observable<Boolean> {
-        TODO("not implemented")
+        return remoteDataSource.removeGroup(groupId)
     }
 
     override fun upInvite(userId: String, invite: Invite): Single<Boolean> {
-        TODO("not implemented")
+        return remoteDataSource.upInvite(userId, invite)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: `upInvite ` to `postInvite `

File: app/src/main/java/vn/asiantech/way/ui/group/GroupActivityUI.kt [hunk 8]
```diff
@@ -13,6 +14,7 @@ class GroupActivityUI : AnkoComponent<GroupActivity> {
     override fun createView(ui: AnkoContext<GroupActivity>) = with(ui) {
         frameLayout {
             lparams(matchParent, matchParent)
+            id = R.id.group_activity_ui_fl_content
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupActivityUI.kt]: `group_activity_ui_fl_content` change to `group_activity_ui_fr_content`

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 9]
```diff
@@ -122,7 +131,46 @@ class GroupRemoteDataSource : GroupDataSource {
...
     override fun removeUserFromGroup(userId: String): Single<Boolean> {
-        TODO("not implemented")
+        val result = SingleSubject.create<Boolean>()
+        HypertrackApi.instance.removeUserFromGroup(userId, BodyAddUserToGroup(null))
+                .doOnSuccess { result.onSuccess(true) }
+                .doOnError { result.onError(it) }
+        return result
+    }
+
+    override fun searchGroup(groupName: String): Observable<List<Group>> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 10]
```diff
@@ -122,7 +131,46 @@ class GroupRemoteDataSource : GroupDataSource {
...
+        val requestRef = firebaseDatabase.getReference("group/$groupId/request/${request.to}")
+        requestRef.setValue(request)
+    }
+
+    override fun postRequestToUser(userId: String, request: Invite) {
+        val requestRef = firebaseDatabase.getReference("user/$userId/request")
+        requestRef.setValue(request)
+    }
+
+    override fun searchUser(name: String): Observable<List<User>> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 11]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
         return remoteDataSource.getInvite(userId)
     }
 
-    override fun getRequest(groupId: String): Observable<Invite> {
-        TODO("not implemented")
+    override fun getGroupRequest(groupId: String): Observable<Invite> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 12]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
+        return remoteDataSource.getGroupRequest(groupId)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 13]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
     }
 
-    override fun upGroupInfo(group: Group): Observable<Boolean> {
-        TODO("not implemented")
+    override fun postGroupInfo(group: Group): Observable<Boolean> {
+        return remoteDataSource.postGroupInfo(group)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 14]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
     }
 
-    override fun changeOwner(groupId: String, newOwner: String): Observable<Boolean> {
-        TODO("not implemented")
+    override fun changeGroupOwner(groupId: String, newOwner: String): Single<Boolean> {
+        return remoteDataSource.changeGroupOwner(groupId, newOwner)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 15]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
     }
 
     override fun removeGroup(groupId: String): Observable<Boolean> {
-        TODO("not implemented")
+        return remoteDataSource.removeGroup(groupId)
     }
 
-    override fun upInvite(userId: String, invite: Invite): Single<Boolean> {
-        TODO("not implemented")
+    override fun postInvite(userId: String, invite: Invite): Single<Boolean> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 16]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
+        return remoteDataSource.postInvite(userId, invite)
     }
 
     override fun removeUserFromGroup(userId: String): Single<Boolean> {
-        TODO("not implemented")
+        return remoteDataSource.removeUserFromGroup(userId)
+    }
+
+    override fun searchGroup(groupName: String): Observable<List<Group>> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 17]
```diff
@@ -27,27 +28,47 @@ class GroupRepository : GroupDataSource {
...
+        return remoteDataSource.searchGroup(groupName)
+    }
+
+    override fun getCurrentRequestOfUser(userId: String): Observable<Invite> {
+        return remoteDataSource.getCurrentRequestOfUser(userId)
+    }
+
+    override fun postRequestToGroup(groupId: String, request: Invite) {
+        remoteDataSource.postRequestToGroup(groupId, request)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: WARNING:

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**